WordPress/Classes/Utility/TaggedManagedObjectID.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+
+/// `TaggedManagedObjectID` is a `NSManagedObjectID` wrapper that also contains the model type of the `NSManagedObjectID`.
+///
+/// By using this strongly typed `NSManagedObjectID`, we can declare APIs that're bound to specific model types, thus
+/// preventing an incorrect object id being used as argument.
+///
+/// Take the following function as an example,
+/// ```
+/// func getPost(fromBlogID blogID: NSManagedObjectID)
+/// ```
+///
+/// We can call the function with any `NSManagedObjectID`: `getPost(fromBlogID: themeObjectID)`. This usage is obviously
+/// incorrect, but we won't be able to catch this error until something goes wrong at rumtime.
+///
+/// However, we can change the declaration to
+/// ```
+/// func getPost(fromBlogID: TaggedManagedObjectID<Blog>)
+/// ```
+///
+/// Now the type `Blog` is built into the function signature, and Swift compiler would report an error if we call it using
+/// `getPost(fromBlogID: themeObjectID)`.
+///
+/// The type name is inspired by the swift-tagged library. The reason we don't use that library is the library exposes
+/// a public initliaser `init(rawValue:)` which is not what we want. With this public initialiser available for all users,
+/// We can't perform the validation described in `init(objectID:)`.
```

Comment:
```suggestion
/// we can't perform the validation described in `init(objectID:)`.
```

The line above ends with a `,` not a `.`